### PR TITLE
reef: docs: warning and remove few docs section for Filestore

### PR DESCRIPTION
**Pull request by zdover23**
Update docs after filestore removal.

Signed-off-by: Nitzan Mordechai <nmordech@redhat.com>
(cherry picked from commit d79f2a81541cfa9e905aef00ad517278c2692c9c)

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by idryomov — COMMENTED** [review]
Please pull https://github.com/ceph/ceph/pull/51159 into this backport.

**Comment by rzarzynski** [comment]
DNM per the Ilya's comment.

File: doc/ceph-volume/lvm/batch.rst [hunk 1]
```diff
@@ -12,8 +12,8 @@ same code path. All ``batch`` does is to calculate the appropriate sizes of all
 volumes and skip over already created volumes.
 
 All the features that ``ceph-volume lvm create`` supports, like ``dmcrypt``,
-avoiding ``systemd`` units from starting, defining bluestore or filestore,
-are supported.
+avoiding ``systemd`` units from starting, defining bluestore,
```
Comment (anthonyeleven) [inline doc/ceph-volume/lvm/batch.rst:15]: s/avoiding/preventing/?

File: doc/ceph-volume/lvm/prepare.rst [hunk 2]
```diff
@@ -19,15 +19,13 @@ play in the Ceph cluster (for example: BlueStore data or BlueStore WAL+DB).
...
-* :ref:`--filestore <ceph-volume-lvm-prepare_filestore>`
 * :ref:`--bluestore <ceph-volume-lvm-prepare_bluestore>`
 
 .. _ceph-volume-lvm-prepare_bluestore:
 
 ``bluestore``
 -------------
-:term:`Bluestore<bluestore>` is the default backend for new OSDs. It
-offers more flexibility for devices than :term:`filestore` does.  Bluestore
+:term:`Bluestore<bluestore>` is the default backend for new OSDs.  Bluestore
```
Comment (anthonyeleven) [inline doc/ceph-volume/lvm/prepare.rst:28]: We usually capitalize the S

File: doc/ceph-volume/lvm/prepare.rst [hunk 3]
```diff
@@ -270,8 +270,7 @@ can be started later (for detailed metadata description see
 Crush device class
 ------------------
 
-To set the crush device class for the OSD, use the ``--crush-device-class`` flag. This will
-work for both bluestore and filestore OSDs::
+To set the crush device class for the OSD, use the ``--crush-device-class`` flag.
```
Comment (anthonyeleven) [inline doc/ceph-volume/lvm/prepare.rst:273]: CRUSH

File: doc/ceph-volume/simple/activate.rst [hunk 4]
```diff
@@ -73,8 +73,7 @@ identify the OSD and its devices, and it will proceed to:
 # mount the device in the corresponding location (by convention this is
   ``/var/lib/ceph/osd/<cluster name>-<osd id>/``)
 
-# ensure that all required devices are ready for that OSD and properly linked,
-regardless of objectstore used (filestore or bluestore). The symbolic link will
-**always** be re-done to ensure that the correct device is linked.
+# ensure that all required devices are ready for that OSD and properly linked. 
+The symbolic link will **always** be re-done to ensure that the correct device is linked.
```
Comment (anthonyeleven) [inline doc/ceph-volume/simple/activate.rst:77]: s/re-done/recreated/

File: doc/man/8/rbd.rst [hunk 5]
```diff
@@ -831,8 +831,7 @@ Per mapping (block device) `rbd device map` options:
   drop discards that are too small.  For bluestore, the recommended setting is
   bluestore_min_alloc_size (currently set to 4K for all types of drives,
   previously used to be set to 64K for hard disk drives and 16K for
-  solid-state drives).  For filestore with filestore_punch_hole = false, the
-  recommended setting is image object size (typically 4M).
+  solid-state drives).
```
Comment (anthonyeleven) [inline doc/man/8/rbd.rst:834]: Suggest HDDs and SSDs

**Review by anthonyeleven — APPROVED** [review]
you know the drill

**Comment by idryomov** [comment]
@anthonyeleven This is a backport PR, so Zac won't be able to act on your suggestions here.  You are probably aware of that (that might be what your "you know the drill" comment refers to, not sure?), but I wanted to mention this anyway since the original PR was merged too quickly and I have some outstanding comments there as well.

**Review by idryomov — COMMENTED** [review]
"doc/man/rbd: resurrect filestore alloc_size note" commit is missing the cherry picked from annotation.

**Comment by anthonyeleven** [comment]
Yep, 3/4 of our doc PRs end up being backports.  Zac often likes to merge anyway and followup.

**Comment by zdover23** [comment]
> @anthonyeleven This is a backport PR, so Zac won't be able to act on your suggestions here. You are probably aware of that (that might be what your "you know the drill" comment refers to, not sure?), but I wanted to mention this anyway since the original PR was merged too quickly and I have some outstanding comments there as well.

As soon as I get back from the grocery store, I'll open a pull request that addresses those outstanding comments.

I treat the docs like a palimpsest, and I know that we can always use a new PR to make improvements that were not made in an earlier PR.

Regarding Anthony's suggestions for improvement being made in a backport PR: Anthony and I had a conversation some time ago in which I let him know that he could comment anywhere that was convenient and I would sort out the matter so that those comments got into the main branch. So, as far as I have anything to say about it, Anthony's cool.

Expect a follow-up PR soon.

**Comment by zdover23** [comment]
The change in d3558c49bf0456a199cf55f73c5832f408462ec5 (https://github.com/ceph/ceph/pull/51159), which restores a sentence that reads "For filestore with filestore_punch_hole = false, the recommended setting is image object size (typically 4M).", is present in the main branch as of the writing of this comment.

This sentence is present in main, and is also present in the reef release branch, the quincy release branch, and the pacific release branch. I know because I just checked.

**Comment by idryomov** [comment]
> This sentence is present in main, and is also present in the reef release branch, the quincy release branch, and the pacific release branch. I know because I just checked.

Right, but this sentence would be removed from the reef branch by the first backported commit in this PR.  The backport of the restoration is now included (as expected), it's just missing `(cherry picked from commit XYZ)` annotation.

**Comment by idryomov** [comment]
> Here you go. I'll merge these and leave them unsquashed if you like.

No, this needs to be redone because the third commit empty (it can't be anything else since you cherry picked right on top of the same cherry pick).  Assuming you are on `wip-doc-2023-04-20-backport-51143-to-reef` branch:
```
$ git rev-parse HEAD~2  # ensure that checkout is correct
45186f16c695d9ee386b0aa29e79e03953494443
$ git reset --hard HEAD~2
$ git cherry-pick -x d3558c49bf0456a199cf55f73c5832f408462ec5
```

**Comment by zdover23** [comment]
> > Here you go. I'll merge these and leave them unsquashed if you like.
> 
> No, this needs to be redone because the third commit is empty (it can't be anything else since you cherry picked right on top of the same cherry pick). Assuming you are on `wip-doc-2023-04-20-backport-51143-to-reef` branch:
> 
> ```
> $ git rev-parse HEAD~2  # ensure that checkout is correct
> 45186f16c695d9ee386b0aa29e79e03953494443
> $ git reset --hard HEAD~2
> $ git cherry-pick -x d3558c49bf0456a199cf55f73c5832f408462ec5
> ```

Thanks for this, @idryomov. I'll try not to be so trigger-happy in the future.

**Review by idryomov — APPROVED** [review]

**Pull request merged**